trace.go

```diff
@@ -28,9 +31,13 @@ type Client interface {
 	// Info returns the unique trace identifier assigned to the current context's trace tree.
 	Info(ctx context.Context) []byte
 
-	// Finish finishes the span in the context with the given labels. Nil labels
-	// should be accepted.
-	Finish(ctx context.Context, labels map[string][]byte) error
+	// Finish finishes the span in the context with the given labels.
+	//
+	// If casual is non-nil, span needs to be finalized with a casual relationship.
+	// Nil labels should be accepted.
+	//
+	// Span should have given start and end time.
+	Finish(ctx context.Context, causal []byte, labels map[string][]byte, start, end time.Time) error
```

Comment:
At this point I think some kind of `FinishParam` would be useful so we're not alway breaking the API if we want to add new args which is likely. We just want to be careful to not go crazy with it.